x-pack/plugins/security_solution/public/timelines/components/timeline/tabs/query/index.tsx

```diff
@@ -197,18 +204,20 @@ export const QueryTabContentComponent: React.FC<Props> = ({
 
   const augumentedColumnHeaders = memoizedGetColumnHeaders(localColumns, browserFields, false);
 
-  const getTimelineQueryFields = () => {
+  const timelineQueryFields = useMemo(() => {
     const columnFields = augumentedColumnHeaders.map((c) => c.id);
 
     return [...columnFields, ...requiredFieldsForActions];
```

Comment:
Not specific to any of your work here, but we need to stop doing array spreads like this, it's so much less performant than a straight `columnFields.concat(requiredFieldsForActions)`.

Reply:
What better time to change this then now? https://github.com/elastic/kibana/pull/180850/commits/e9fd124bf6c31da90fbef585048b46c813193410